backend/logic/users/users.js

```diff
@@ -29,157 +17,128 @@ var insertStudentUser = async (username, password, email, id) => {
+    );
+    await connection.query("DELETE FROM students WHERE student_id = ?;", [
+      student_id[0].student_id
+    ]);
     await connection.query("DELETE FROM users WHERE username = ?;", [username]);
     await connection.commit();
     connection.release();
     return true;
   } catch (error) {
-    console.error(error);
     connection.rollback();
     connection.release();
-    throw new Error(false);
+    throw error;
   }
 };
 
 var insertAdminUser = async (username, password, email, id) => {
   // Connect to database
-  let error = false;
   // Check for invalid formatting
-  //todo, handle errors after formatting configured to throw errors
-  if (!format.verifyUsername(username)) {
-    error = "invalid format username";
-  } else if (!format.verifyPassword(password)) {
-    error = "invalid format password";
-  } else if (!format.verifyEmail(email)) {
-    error = "invalid format email";
-  } else if (!format.verifyAdminId(id)) {
-    error = "invalid format id";
-  }
-
-  if (!error == false) {
-    console.error(error);
-    throw new Error(error);
-  }
+  format.verifyUsername(username);
+  format.verifyPassword(password);
+  format.verifyEmail(email);
+  format.verifyId(id);
 
   let connection = await mysql.getNewConnection();
 
   // Hash the password
   let hash = hasher.hashPass(password);
   try {
     await connection.beginTransaction();
-    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [username, email, hash]);
-    await connection.query("INSERT INTO staff_members VALUES(?, ?);", [id, username]);
+    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [
+      username,
+      email,
+      hash
+    ]);
+    await connection.query("INSERT INTO staff_members VALUES(?, ?);", [
+      id,
+      username
+    ]);
     await connection.commit();
     connection.release();
     return true;
   } catch (error) {
     connection.rollback();
     connection.release();
-    console.error(error);
-    throw new Error(false);
+    throw error;
   }
 };
 
 var deleteAdminUser = async username => {
-  if (!format.verifyUsername(username)) {
-    let error = "invalid username";
-    console.error(error);
-    throw new Error(error);
-  }
+  format.verifyUsername(username);
 
-  let connection;
-  try {
-    connection = await mysql.getNewConnection();
-  } catch (error) {
-    console.error(error);
-    throw new Error("failed to establish connection with database");
-  }
+  let connection = await mysql.getNewConnection();
 
   try {
     await connection.beginTransaction();
-    let staff_id = await connection.query("SELECT staff_id FROM staff_members WHERE username = ?;", username);
-    await connection.query("DELETE FROM staff_members WHERE staff_id = ?;", [staff_id[0].staff_id]);
+    let staff_id = await connection.query(
+      "SELECT staff_id FROM staff_members WHERE username = ?;",
+      username
+    );
+    await connection.query("DELETE FROM staff_members WHERE staff_id = ?;", [
```

Comment:
Pretty sure we're able to reduce these 2 queries into a single one without causing any issues

DELETE FROM staff_members WHERE staff_id IN (SELECT staff_id FROM staff_members WHERE username = ?);